app/controllers/auth_failures_controller.rb

```diff
@@ -8,14 +8,15 @@ def failure
     when :identity
       handle_failure_then_redirect_to qualifications_root_path
     when :dfe
-      handle_failure_then_redirect_to check_records_sign_in_path
+      return redirect_to(check_records_dsi_sign_out_path(id_token_hint: session[:id_token])) if sessionexpired?
+
+      handle_failure_then_redirect_to check_records_sign_in_path(oauth_failure: true)
```

Comment:
I wonder if we can simplify here by just sending the user to the dsi_sign_out_path regardless of what the failure is? Would that fail if session[:id_token] is nil?

Reply:
Without a token the user flow will end at an unbranded DSI page with no navigation to return to the service. At least this is the case on the dev, test and preprod sign in services.